deltachat-server/rpc-client.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+import asyncio
+import json
+
+
+class Deltachat:
+    def __init__(self, process):
+        self.process = process
+        self.event_queue = asyncio.Queue()
+        self.id = 0
+        self.reader_task = asyncio.create_task(self.reader_loop())
+
+        # Map from request ID to asyncio.Event.
+        # When event happens, it is replaced with a response.
+        self.request_events = {}
+
+    async def reader_loop(self):
+        while True:
+            line = await self.process.stdout.readline()
+            response = json.loads(line)
+            if "id" in response:
+                response_id = response["id"]
+                event = self.request_events[response_id]
+                self.request_events[response_id] = response
+
+                # Notify that response arrived.
+                event.set()
+            else:
+                if response["method"] == "event":
+                    # An event notification.
+                    await self.event_queue.put(response["params"])
+                continue
+
+    async def get_next_event(self):
+        return await self.event_queue.get()
+
+    def __getattr__(self, attr):
+        async def method(*args):
+            self.id += 1
+            request_id = self.id
+
+            request = {
+                "jsonrpc": "2.0",
+                "method": attr,
+                "params": list(args),
+                "id": self.id,
+            }
+            data = (json.dumps(request) + "\n").encode()
+            self.process.stdin.write(data)
+            event = asyncio.Event()
+            self.request_events[request_id] = event
+            await event.wait()
+            response = self.request_events[request_id]
+            del self.request_events[request_id]
+            if "result" in response:
+                return response["result"]
+            raise ValueError(response["error"])
+
+        return method
+
+
+async def main():
+    proc = await asyncio.create_subprocess_exec(
+        "./deltachat-server",
+        stdin=asyncio.subprocess.PIPE,
+        stdout=asyncio.subprocess.PIPE,
+    )
+    deltachat = Deltachat(proc)
+
+    async def event_loop():
+        while event := await deltachat.get_next_event():
+            print(event)
```

Comment:
I'm confused, where does this print go to?  Doesn't this go straight back to the `deltachat-server`?